server/src/test/java/org/elasticsearch/cluster/metadata/MetadataTests.java

```diff
@@ -878,6 +878,44 @@ public void testFindMappingsWithFilters() throws IOException {
         }
     }
 
+    public void testOldestIndexComputation() {
+        Metadata metadata = buildIndicesWithVersions(
+            new Version[] { Version.V_7_0_0, Version.CURRENT, Version.fromId(Version.CURRENT.id + 1) }
+        ).build();
+
+        assertEquals(Version.V_7_0_0, metadata.oldestIndexVersion());
+
+        Metadata.Builder b = Metadata.builder();
+        assertEquals(Version.CURRENT, b.build().oldestIndexVersion());
+
+        Throwable ex = expectThrows(
+            IllegalArgumentException.class,
+            () -> buildIndicesWithVersions(new Version[] { Version.V_7_0_0, Version.V_EMPTY, Version.fromId(Version.CURRENT.id + 1) })
+                .build()
+        );
+
+        assertEquals("[index.version.created] is not present in the index settings for index with UUID [null]", ex.getMessage());
+    }
+
+    private Metadata.Builder buildIndicesWithVersions(Version[] indexVersions) {
+
+        final List<Index> indices = new ArrayList<>();
+        int lastIndexNum = randomIntBetween(9, 50);
+        Metadata.Builder b = Metadata.builder();
+        for (int k = 0; k < indexVersions.length; k++) {
```

Comment:
Think you can use a foreach here?

```suggestion
        for (Version indexVersion : indexVersions) {
```